python/kvikio/kvikio/_lib/libnvcomp_ll.pyx

```diff
@@ -371,20 +372,26 @@ class nvCompBatchAlgorithmLZ4(nvCompBatchAlgorithm):
 
     HEADER_SIZE_BYTES: size_t = sizeof(uint32_t)
 
-    def __init__(self, data_type: int = 0, has_header: bool = True):
+    def __init__(self, data_type: int = None, has_header: bool = True):
```

Comment:
```suggestion
    def __init__(self, data_type: int | None = None, has_header: bool = True):
```

Reply:
Think Cython currently doesn't handle `typing.Union`s ( https://github.com/cython/cython/pull/6206 ) and probably won't. Though it might handle the special case with `None` in the future ( https://github.com/cython/cython/issues/6254 ), it doesn't appear to be included yet

Reading [this Cython FAQ]( https://cython.readthedocs.io/en/latest/src/userguide/faq.html#why-does-a-function-with-cdef-d-parameters-accept-none ) it suggests to pick one of...

* `func(x: list = None)` (as Bradley has)
* `func(x: Optional[list])` (from locally testing found it needed to be `func(x: Optional[list] = None)`)

So which do you prefer?

Reply:
No strong opinion between the two, let's god with Bradley's approach :)